### ISSUE TITLE
re2 compilation fails due missing -std=c++11 flag

### ISSUE THREAD

**Issue by balcruz** [matehat/cqerl#90]
Environment: Linux  4.4.0-1-amd64 #1 SMP Debian 4.4.6-1 (2016-03-17) x86_64 GNU/Linux
Erlang: R18.2
cqerl version: master (re2 v1.3.1)

**Steps to reproduce:**

1. Clone cqerl repo
`
 git clone https://github.com/matehat/cqerl.git
`

2. Compile
` 
make
`
After compile the following error is displayed:

> Compiling /home/berna/cqerl/deps/re2/c_src/re2_nif.cpp
In file included from /usr/include/c++/5/mutex:35:0,
                 from c_src/re2/re2/re2.h:184,
                 from /home/berna//cqerl/deps/re2/c_src/re2_nif.cpp:8:
/usr/include/c++/5/bits/c++0x_warning.h:32:2: error: #error This file requires compiler and library support for the ISO C++ 2011 standard. This support must be enabled with the -std=c++11 or -std=gnu++11 compiler options.
 #error This file requires compiler and library support \
  ^
In file included from /home/berna/cqerl/deps/re2/c_src/re2_nif.cpp:8:0:
c_src/re2/re2/re2.h:734:16: error: 'once_flag' in namespace 'std' does not name a type
   mutable std::once_flag rprog_once_;
                ^
c_src/re2/re2/re2.h:735:16: error: 'once_flag' in namespace 'std' does not name a type
   mutable std::once_flag num_captures_once_;
                ^
c_src/re2/re2/re2.h:736:16: error: 'once_flag' in namespace 'std' does not name a type
   mutable std::once_flag named_groups_once_;
                ^
c_src/re2/re2/re2.h:737:16: error: 'once_flag' in namespace 'std' does not name a type
   mutable std::once_flag group_names_once_;
                ^
ERROR: compile failed while processing /home/berna/cqerl/deps/re2: rebar_abort
Makefile:10: recipe for target 'compile' failed
make: *** [compile] Error 1

Looking at [re2](https://github.com/tuncer/re2/commit/cac4f871c06340bc5745fa1bf84fa56893306d48) I noticed the -std=c++11 flag was added to re2  v1.4.0.

Do you think re2 could be upgraded to v1.4.0 at rebar.config (see below) without break anything? 
```
diff --git a/rebar.config b/rebar.config
index 144f403..d3fd585 100644
--- a/rebar.config
+++ b/rebar.config
@@ -6,5 +6,5 @@
   {semver, ".*", {git, "https://github.com/nebularis/semver.git", "c7d509"}},
   {uuid, ".*", {git, "https://github.com/okeuday/uuid.git", {tag, "v1.4.1"}}},
   {pooler, ".*", {git, "https://github.com/seth/pooler.git", {tag, "1.5.0"}}},
-  {re2, ".*", {git, "https://github.com/tuncer/re2.git", {tag, "v1.3.1"}}}
+  {re2, ".*", {git, "https://github.com/tuncer/re2.git", {tag, "v1.4.0"}}}
 ]}.
```

Thanks,
Bernardo

**Comment by matehat** [comment]
Have you tried it?

**Comment by balcruz** [comment]
Yes, it compiled without issues.

**Comment by matehat** [comment]
Fixed in cfd37e77ca3d29735a6a8243f5686cdb187b5994. Thanks!
